lib/Driver/DarwinToolChains.cpp

```diff
@@ -723,6 +738,10 @@ toolchains::Darwin::constructInvocation(const DynamicLinkJobAction &job,
 
   addArgsToLinkARCLite(Arguments, context);
 
+  if (context.Args.hasArg(options::OPT_lto)) {
```

Comment:
Why not represent this in the `JobAction` instead?  When the job is constructed, we should be able to identify at that point if this is going to be a LTO job or not.

Reply:
Added a flag that indicates that the job should perform LTO or not.